### ISSUE TITLE
routef not working for multiple params

### ISSUE THREAD

**Issue by Potmos** [giraffe-fsharp/Giraffe#395]
I am unable to make routef work for multiple param while following the documentation.

I am following https://github.com/giraffe-fsharp/Giraffe/blob/master/DOCUMENTATION.md

I tried the 
`routef "/foo/%s/%s/%s" (fun s1, s2, s3 -> text (sprintf "%s%s%s" s1 s2 s3))`
And get the error "Unexpected symbol ',' in lambda expression. Expected '->' or other token.", then I changed to 
`routef "/foo/%s/%s/%s" (fun s1 s2 s3 -> text (sprintf "%s%s%s" s1 s2 s3))`
"This expression was expected to have type 'HttpFuncResult' but here has type 'HttpFunc -> 'a -> HttpFuncResult'"

I tried the more complex 
```
let fooHandler (first : string)
               (last  : string)
               (age   : int)
               : HttpHandler =
    fun (next : HttpFunc) (ctx : HttpContext) ->
        (sprintf "First: %s, Last: %s, Age: %i" first last age
        |> text) next ctx

let webApp =
    choose [
        routef "/foo/%s/%s/%i" fooHandler
        routef "/bar/%O" (fun guid -> text (guid.ToString()))

        // If none of the routes matched then return a 404
        RequestErrors.NOT_FOUND "Not Found"
    ]
```
and get the error "The type 'string * string * 'a' does not match the type 'string'".
I've been away from F# for a while so I guess I may be doing something wrong, but since I am just following the documentation I can't tell what.

**Comment by Potmos** [comment]
Okay I found an older example and was able to make it work changing to a tupple function.

`routef "/foo/%s/%s" (fun (x, y) -> text (sprintf "%s - %s" x y ))`

**Comment by slang25** [comment]
@Potmos Thanks for raising this, the documentation code was broken, I've fixed here: https://github.com/giraffe-fsharp/Giraffe/commit/0e811cf4000b272de2522a99bbea60027b670ef6

**Comment by Potmos** [comment]
> @Potmos Thanks for raising this, the documentation code was broken, I've fixed here: [0e811cf](https://github.com/giraffe-fsharp/Giraffe/commit/0e811cf4000b272de2522a99bbea60027b670ef6)

Thank you for the fast response this is really helpfull. :)
